azure/scope/identity.go

```diff
@@ -136,44 +139,50 @@ func (p *ManagedControlPlaneCredentialsProvider) GetAuthorizer(ctx context.Conte
 
 // GetAuthorizer returns an Azure authorizer based on the provided azure identity and cluster metadata.
 func (p *AzureCredentialsProvider) GetAuthorizer(ctx context.Context, resourceManagerEndpoint, activeDirectoryEndpoint string, clusterMeta metav1.ObjectMeta) (autorest.Authorizer, error) {
-	var spt *adal.ServicePrincipalToken
+	var authErr error
+	var cred azcore.TokenCredential
```

Comment:
Any particular reason for declaring these variables explicitly vs using shorthand in the switch case?  

e.g

```
cred, err := azidentity.NewManagedIdentityCredential(&options)
```

Reply:
Yes. If I don't do this then I can't use them on lines https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2748/files#diff-b6ab207506064afcbd1d777c1c7296e2953d2653a8843bfb39107af5cf5b0a87R174-R177.

The alternative would be to duplicate those lines in each `case` branch, which I tried to avoid.